cli/status/status.go

```diff
@@ -1,57 +1,185 @@
+	ReadOnly        string            `mapstructure:"read_only"`
+	BoxStatus       string            `mapstructure:"box_status"`
 }
 
 // Status writes the status as a table.
 func Status(runningCtx running.RunningCtx, opts StatusOpts) error {
 	ts := table.NewWriter()
 	ts.SetOutputMirror(os.Stdout)
-	ts.AppendHeader(table.Row{"INSTANCE", "STATUS", "PID", "MODE"})
+	ts.AppendHeader(
+		table.Row{"INSTANCE", "STATUS", "PID", "MODE", "CONFIG", "BOX", "UPSTREAM"})
+
+	instances := map[string]map[string]interface{}{}
+	instancesAlerts := map[string][]string{}
+
+	PrintYellow := color.New(color.FgYellow).SprintFunc()
+	PrintRed := color.New(color.FgRed).SprintFunc()
 
 	for _, run := range runningCtx.Instances {
-		row := []interface{}{}
 		fullInstanceName := running.GetAppInstanceName(run)
 		procStatus := running.Status(&run)
 
-		row = append(row, fullInstanceName)
-		row = append(row, procStatus.ColorSprint(procStatus.Status))
+		if _, exists := instances[fullInstanceName]; !exists {
+			instances[fullInstanceName] = map[string]interface{}{
+				"STATUS": "",
+				"PID":    nil,
+				"MODE":   "",
+				"CFG":    "--",
+				"BOX":    "--",
+				"UPS":    "--",
+			}
+		}
+
+		instances[fullInstanceName]["STATUS"] = procStatus.ColorSprint(procStatus.Status)
 		if procStatus.Code == process_utils.ProcessRunningCode {
-			row = append(row, procStatus.PID)
+			instances[fullInstanceName]["PID"] = procStatus.PID
 		}
 
 		conn, err := connector.Connect(connector.ConnectOpts{
 			Network: "unix",
 			Address: run.ConsoleSocket,
 		})
+
 		if err == nil {
-			res, err := conn.Eval("return (type(box.cfg) == 'function') or box.info.ro",
-				[]any{}, connector.RequestOpts{})
+			var instanceState InstanceState
+			isParsed := false
+			res, err := conn.Eval(replicationInfoLuaScript, []any{},
+				connector.RequestOpts{})
 			if err == nil && len(res) != 0 {
-				mode := res[0].(bool)
-				mode_str := "RO"
-				if !mode {
-					mode_str = "RW"
+				isParsed = mapstructure.Decode(res[0], &instanceState) == nil
+			}
+			if !isParsed {
+				continue
+			}
+			instances[fullInstanceName]["MODE"] = instanceState.ReadOnly
+			instances[fullInstanceName]["CFG"] = instanceState.ConfigInfo.Status
+			instances[fullInstanceName]["BOX"] = instanceState.BoxStatus
+
+			for _, repl := range instanceState.ReplicationInfo {
+				if repl.Upstream == nil {
+					continue
+				}
+				fullInstanceUpstreamName := ""
+				if run.SingleApp {
+					fullInstanceUpstreamName = run.AppName
+				} else {
+					fullInstanceUpstreamName = run.AppName + ":" + repl.Name
+				}
+				if _, exists := instances[fullInstanceUpstreamName]; !exists {
+					instances[fullInstanceUpstreamName] =
+						map[string]interface{}{
+							"STATUS": "",
+							"PID":    nil,
+							"MODE":   "",
+							"CFG":    "--",
+							"BOX":    "--",
+							"UPS":    "--",
+						}
+				}
+				instances[fullInstanceUpstreamName]["UPS"] = repl.Upstream.Status
```

Comment:
Should we really report a connetivity problem in the A->B replication data flow that is detected on B in the A's row?

For me it looks like a problem that definitely affects B, but it is not necessary that something is wrong with A. Maybe just a network problem between them.

Reply:
> Should we really report a connetivity problem in the A->B replication data flow that is detected on B in the A's row?

I actually don't know.

>For me it looks like a problem that definitely affects B, but it is not necessary that something is wrong with A. Maybe just a network problem between them.

Maybe you are right, but it seems to me that if we have any alerts in `box.info.replication.[*].upstream`, then there are none in the instance component itself

Reply:
Am I understand you right: taking my example with broken A->B replication you're worrying that nothing said in A's row in the `tt status` output?

Let's assume that we report an upstream's problem according to my proposal: in B's row.

If the problem is purely about a network between A and B, then we see something like `disconnected` in `UPSTREAM` column in B's row. B is disconnected from some of its upstreams. OK, seems intuitive.

If A is in a bad state (the process is terminated, for example), we see this information in A's row and see `disconnected` in B's row. It also looks intuitive.

----

TBH, I have ideas like reporting `1/3 disconnected` or something of this kind in `UPSTREAM`, but I think that I shouldn't raise a discussion around such details in the scope of this pull request.

Reply:
We discussed it with Maksim. Results:

1. Agreed to report B's upstream problem in B's row.
2. Agreed to report simple `disconnected` or similar status if different ones are present. To be precise: report first non-follow status (or just follow if all the upstreams are in the follow status).
3. Agreed on the following `--details` message (or something similar):

   > [upstream][warning] replication from "app3:storage-001-b" is in "disconnected" status: <..error message..>

Reply:
Done.